bson/src/main/org/bson/codecs/configuration/mapper/MethodModel.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.configuration.mapper;
+
+import com.fasterxml.classmate.members.ResolvedMethod;
+import org.bson.codecs.configuration.CodecRegistry;
+
+/**
+ * Represents the metamodel of a method on a class mapped by a ClassModel
+ */
+public class MethodModel {
```

Comment:
This doesn't seem complete... I assumed it would inherit from MappedType just like FieldModel.